### PR TITLE
chore(lint): updates golangci-lint to latest version

### PR DESCRIPTION
**Pull request by bartoszmajsak**
### Fixes

- removed irrelevant `//nolint` comments (see config updates below)
- converted unused func params to blank identifier when impl does not need it

### New linters

`perfsprint`

- disabled fmt.Sprintf checks for one argument string message, as it's been used in multiple places. I assume it's to favor readability and ease for further extension if more args needs
- adjusted errors creation to use errors.New instead of fmt.Errorf when no-arg strings are used as messages

### Configuration updates

- configured revive to not complain about dot-imports for testing libraries
- adjusted configuration due to changes in the structure (see `skip-dirs`)
- some deprecated linters were removed from golangci-lint toolchain so they are now gone from the config too

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] ~Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).~
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -30,38 +28,38 @@ linters-settings:
...
     - tagliatelle
-    - varcheck # deprecated
     - varnamelen # doesnot allow shorter names like c,k etc. But golang prefers short named vars.
     - wsl # [too strict and mostly code is not more readable] whitespace linter forces you to use empty lines
     - wrapcheck # check if this is required. Prevents direct return of err.
 
     # Need to check
     - nlreturn # [too strict and mostly code is not more readable] checks for a new line before return and branch statements to increase code clarity
-    - goerr113 # [too strict] checks the errors handling expressions
+    - err113 # [too strict] checks the errors handling expressions
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply .golangci.yml:62]: No, it was replaced by this one

File: .golangci.yml [hunk 2]
```diff
@@ -71,6 +69,8 @@ linters:
     - godox    # https://github.com/opendatahub-io/opendatahub-operator/issues/699
 
 issues:
+  exclude-dirs:
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply .golangci.yml:72]: Good question. I just it moved to the new config block. Maybe we should narrow the skip to auto generated files instead.  But I wonder if that's in scope of this PR. Maybe we should capture it as a task to review linter findings and decide which one to still exclude and also review the exclusion rules such as this. WDYT?

**Review by AjayJagan — APPROVED** [review]

File: .golangci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply .golangci.yml:72]: we have a ticket https://issues.redhat.com/browse/RHOAIENG-481 maybe we should extend it to the new findings after this PR?

File: .golangci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply .golangci.yml:62]: right,  in [v1.58.0](https://golangci-lint.run/product/changelog/#v1580) , "goerr113 is replaced by err113"

**Review by zdtsw — APPROVED** [review]

File: tests/e2e/dsc_creation_test.go [hunk 3]
```diff
@@ -12,7 +13,7 @@ import (
 	operatorv1 "github.com/openshift/api/operator/v1"
 	"github.com/stretchr/testify/require"
 	autoscalingv1 "k8s.io/api/autoscaling/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (ykaliuta) [inline tests/e2e/dsc_creation_test.go:16]: should it be `apierrs` like in other places?

Comment (bartoszmajsak) [reply tests/e2e/dsc_creation_test.go:16]: Seen both. Will check other places and align in a follow up PR.

Comment (bartoszmajsak) [reply tests/e2e/dsc_creation_test.go:16]: @ykaliuta I opened a follow-up PR #1078 1078

**Comment by ykaliuta** [comment]
I get 
```
diff --git a/controllers/status/reporter.go b/controllers/status/reporter.go
index e5b3679ff326..2e70298616c7 100644
--- a/controllers/status/reporter.go
+++ b/controllers/status/reporter.go
@@ -1,4 +1,4 @@
-//nolint:ireturn //reason: return T which is expected to be satisfying client.Object interface
+
 package status
 
 import (
```
after `make lint` now

**Comment by bartoszmajsak** [comment]
> I get
> 
> ```
> diff --git a/controllers/status/reporter.go b/controllers/status/reporter.go
> index e5b3679ff326..2e70298616c7 100644
> --- a/controllers/status/reporter.go
> +++ b/controllers/status/reporter.go
> @@ -1,4 +1,4 @@
> -//nolint:ireturn //reason: return T which is expected to be satisfying client.Object interface
> +
>  package status
>  
>  import (
> ```
> 
> after `make lint` now

@ykaliuta I hope this https://github.com/opendatahub-io/opendatahub-operator/pull/1085 fixed the problem.

**Comment by ykaliuta** [comment]
> 
> @ykaliuta I hope this #1085 fixed the problem.

yep. Thank you!

**Pull request merged**